// Copyright 2020 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa.common;

option java_package = "org.wfanet.common";
option java_multiple_files = true;

// Public keys of the ElGamal cipher.
message ElGamalPublicKey {
  // Public key generator `g`.
  bytes generator = 1;
  // Public key element `y = g^x`, where `g` is the generator and `x` is the
  // secret key.
  bytes element = 2;
}

// Keys of the ElGamal cipher.
message ElGamalKeyPair {
  // Secret key of the ElGamal cipher.
  bytes secret_key = 1;
  // Public key of the ElGamal cipher.
  ElGamalPublicKey public_key = 2;
}